battle.go

```diff
@@ -6,19 +6,18 @@ import (
 
 // A Pokemon battle. Enforces rules of the battle, and queries `Agent`s for turns.
 type Battle struct {
-	Weather      int  // one of the 6 in-battle weather conditions
-	ShiftSet     bool // shift or set battle style for NPC trainer battles
-	State        BattleState
-	Parties      []*Party
-	Agents       []*Agent
-	agentParties map[int]int // Maps agent indexes to party indexes
-	teams        [][]int     // An array of teams, which are arrays of Agents. Used to derive allies and opponents
+	Weather  int  // one of the 6 in-battle weather conditions
+	ShiftSet bool // shift or set battle style for NPC trainer battles
+	State    BattleState
+	Parties  []*Party
+	teams    [][]int // An array of teams, which are arrays of Party ids. Used to derive allies and opponents
```

Comment:
(nit) Can team also just be a field on Party? If it's tracking the Agent presumably it can also track the team that the Agent is on?